modin/core/dataframe/base/dataframe/utils.py

```diff
@@ -48,3 +53,71 @@ class JoinType(Enum):  # noqa: PR01
     LEFT = "left"
     RIGHT = "right"
     OUTER = "outer"
+
+
+def join_columns(
+    left: pandas.Index,
+    right: pandas.Index,
+    left_on: Union[Iterable[IndexLabel], IndexLabel],
+    right_on: Union[Iterable[IndexLabel], IndexLabel],
```

Comment:
In pandas they are defined differently: https://github.com/pandas-dev/pandas/blob/v1.5.3/pandas/core/frame.py#L10074-L10107
```suggestion
    left_on: IndexLabel | None,
    right_on: IndexLabel | None,
```

Reply:
I think the suggestion should be like this

```python
left_on: IndexLabel,
right_on: IndexLabel,
```

because left_on and right_on are required args.